File: api/players.py
import json
from datetime import datetime
import sys
import os

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
from api.lib.mongodb import get_database

CORS_HEADERS = {
    'Access-Control-Allow-Origin': '*',
    'Access-Control-Allow-Methods': 'GET, POST, DELETE, OPTIONS',
    'Access-Control-Allow-Headers': 'Content-Type',
    'Content-Type': 'application/json'
}

def handler(request):
    """Vercel serverless handler for players API"""
    
    # Handle CORS preflight
    if request.method == 'OPTIONS':
        return {
            'statusCode': 200,
            'headers': CORS_HEADERS,
            'body': ''
        }
    
    db = get_database()
    
    if db is None:
        return {
            'statusCode': 503,
            'headers': CORS_HEADERS,
            'body': json.dumps({
                'success': False,
                'error': 'Database not configured. Please set MONGODB_URI in environment.',
                'players': []
            }, default=str)
        }
    
    try:
        if request.method == 'GET':
            players_collection = db['players']
            all_players = list(players_collection.find({}).sort('joinedAt', -1))
            
            # Convert ObjectId to string
            for player in all_players:
                player['_id'] = str(player['_id'])
            
            return {
                'statusCode': 200,
                'headers': CORS_HEADERS,
                'body': json.dumps({
                    'success': True,
                    'players': all_players
                }, default=str)
            }
        
        elif request.method == 'POST':
            # Parse request body
            try:
                body = json.loads(request.body) if isinstance(request.body, str) else json.loads(request.body.decode('utf-8'))
            except:
                body = {}
            
            name = body.get('name', '').strip()
            
            if not name:
                return {
                    'statusCode': 400,
                    'headers': CORS_HEADERS,
                    'body': json.dumps({
                        'success': False,
                        'error': 'Name is required'
                    })
                }
            
            players_collection = db['players']
            
            new_player = {
                'name': name,
                'joinedAt': datetime.utcnow(),
                'score': 0,
                'isActive': True
            }
            
            result = players_collection.insert_one(new_player)
            new_player['_id'] = str(result.inserted_id)
            
            return {
                'statusCode': 201,
                'headers': CORS_HEADERS,
                'body': json.dumps({
                    'success': True,
                    'player': new_player
                }, default=str)
            }
        
        elif request.method == 'DELETE':
            players_collection = db['players']
            players_collection.delete_many({})
            
            return {
                'statusCode': 200,
                'headers': CORS_HEADERS,
                'body': json.dumps({
                    'success': True,
                    'message': 'All players removed'
                })
            }
        
        else:
            return {
                'statusCode': 405,
                'headers': CORS_HEADERS,
                'body': json.dumps({
                    'success': False,
                    'error': 'Method not allowed'
                })
            }
    
    except Exception as e:
        print(f'Players API error: {str(e)}')
        return {
            'statusCode': 500,
            'headers': CORS_HEADERS,
            'body': json.dumps({
                'success': False,
                'error': str(e)
            })
        }

File: api/players.js
const clientPromise = require('./lib/mongodb.js');

module.exports = async function handler(req, res) {
  // Enable CORS
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET, POST, DELETE, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }

  if (!clientPromise) {
    return res.status(503).json({ 
      success: false, 
      error: 'Database not configured. Please set MONGODB_URI in .env file.' 
    });
  }

  const client = await clientPromise;
  
  if (!client) {
    return res.status(503).json({ 
      success: false, 
      error: 'Database connection failed. Please check your MONGODB_URI.',
      players: []
    });
  }

  const db = client.db('puzzlegame');
  const players = db.collection('players');

  try {
    if (req.method === 'GET') {
      // Get all players
      const allPlayers = await players.find({}).sort({ joinedAt: -1 }).toArray();
      return res.status(200).json({ success: true, players: allPlayers });
    }

    if (req.method === 'POST') {
      // Add new player
      const { name } = req.body;
      
      if (!name || name.trim() === '') {
        return res.status(400).json({ success: false, error: 'Name is required' });
      }

      const newPlayer = {
        name: name.trim(),
        joinedAt: new Date(),
        score: 0,
        isActive: true
      };

      const result = await players.insertOne(newPlayer);
      return res.status(201).json({ 
        success: true, 
        player: { ...newPlayer, _id: result.insertedId } 
      });
    }

    if (req.method === 'DELETE') {
      // Clear all players (for testing)
      await players.deleteMany({});
      return res.status(200).json({ success: true, message: 'All players removed' });
    }

    return res.status(405).json({ success: false, error: 'Method not allowed' });
  } catch (error) {
    console.error('Players API error:', error);
    return res.status(500).json({ success: false, error: error.message });
  }
}
